ZFSin/zfs/module/zfs/zfs_vnops.c

```diff
@@ -2116,18 +2116,20 @@ zfs_remove(vnode_t *dvp, char *name, cred_t *cr, caller_context_t *ct,
 		mutex_exit(&zp->z_lock);
 		vnode_pager_setsize(vp, 0);
 		VN_RELE(vp);
+
 		/*
 		 * Call recycle which will call vnop_reclaim directly if it can
 		 * so tell reclaim to not do anything with this node, so we can
 		 * release it directly. If recycle/reclaim didn't work out, defer
 		 * it by placing it on the unlinked list.
 		 */
-
 		zp->z_fastpath = B_TRUE;
-		if (vnode_recycle(vp) == 1) {
+		if (vnode_recycle(vp) == 0) {
```

Comment:
vnode_recycle always return 0 at the moment. On success we zfs_znode_delete it. Right?